tests/ApplySecureHeadersTest.php

```diff
@@ -300,65 +268,27 @@ public function testCSPRO()
         $this->assertBaseHeadersPresent($headers);
     }
 
-    /**
-     * Assert base headers are present given an array of headers.
-     *
-     * @param array<string, string[]> $headers
-     * @return void
-     */
-    private function assertBaseHeadersPresent(array $headers)
+    public function testProtectedCookies()
     {
-        $this->assertArrayHasKey('x-permitted-cross-domain-policies', $headers);
-        $this->assertArrayHasKey('x-content-type-options', $headers);
-        $this->assertArrayHasKey('expect-ct', $headers);
-        $this->assertArrayHasKey('referrer-policy', $headers);
-        $this->assertArrayHasKey('x-xss-protection', $headers);
-        $this->assertArrayHasKey('x-frame-options', $headers);
-    }
-
-    /**
-     * Apply SecureHeaders from the given config to a Response.
-     *
-     * @param Response $response
-     * @param array $configMap
-     * @return Response
-     */
-    private function applySecureHeadersWithConfig(Response $response, array $configMap = [])
-    {
-        // convert the configMap dot keys into deeply nested array
+        // configuration
         $configMap = [
-            'secure-headers' => !empty($configMap)
-                ? array_merge_recursive(...array_map(
-                    [self::class, 'nonMutatingDataFill'],
-                    array_fill(0, count($configMap), []),
-                    array_keys($configMap),
-                    $configMap
-                ))
-                : []
+            'protectedCookies' => [
+                'add' => [
+                    'names' => ['lol'],
+                    'substrings' => ['whut'],
+                ],
+                'remove' => [
+                    'names' => [],
+                    'substrings' => [],
+                ],
+            ],
         ];
 
-        /** @var Repository|\PHPUnit_Framework_MockObject_MockObject $config */
-        $config = $this->createMock(Repository::class);
-
-        $config->expects($this->any())
-            ->method('get')
-            ->with($this->anything())
-            ->will($this->returnCallback(function (string $key, $default) use ($configMap) {
-                return Arr::get($configMap, $key, $default);
-            }));
-
-        $secureHeaders = new SecureHeaders();
-        $secureHeaders->errorReporting(false);
-        $middleware = new ApplySecureHeaders($config, $secureHeaders);
+        $result  = $this->applySecureHeadersWithConfig(new Response, $configMap);
+        $headers = $result->headers->all();
 
-        return $middleware->handle(new Request, function () use ($response) {
-            return $response;
-        });
-    }
+        dd($result);
```

Comment:
@aidantwoods I'm not seeing anything in this dump after using `protectedCookie`:

https://github.com/mikefrancis/laravel-secureheaders/pull/38/files#diff-d7aa6dfd3694cac6b7ce2fe2b35c3a8cR134

Am I doing something wrong?

Reply:
Is the cookie set anywhere? :) The purpose of `protectedCookie` behaviour is to make sure the named cookies don't go out without flags like `secure`, but if the cookie is never set then there's nothing to add the flags to.

If you set cookies `lol` and `whut` (e.g. add the headers `Set-Cookie: lol=foo`, `Set-Cookie: whut=bar`) then the configured flags should be added.

Reply:
There's nothing being set, no. Not sure if this something to do with how the test applies the middleware to the request in `applySecureHeadersWithConfig()` 🤔

Reply:
You need to set some cookies that will match to observe an effect :) If no cookies are set then nothing happens.

E.g. if the cookie named `foo` is configured as a protected cookie and the following header is in the original response

```
Set-Cookie: foo=bar
```

then after applying SecureHeaders you should instead see the header:

```
Set-Cookie: foo=bar; Secure; HttpOnly; SameSite=Lax
```

Reply:
Of course 🤦 sorry, had a long night last night!